assets/js/googlesitekit/datastore/user/user-info.test.js

```diff
@@ -364,5 +364,53 @@ describe( 'core/user userInfo', () => {
 				expect( global[ userDataGlobal ] ).not.toEqual( undefined );
 			} );
 		} );
+
+		describe( 'getAccountChooserURL', () => {
+			it( 'throws an error if a destinationURL is not given', () => {
+				expect( () => {
+					registry.select( CORE_USER ).getAccountChooserURL();
+				} ).toThrow( 'destinationURL is required' );
+			} );
+
+			it( 'returns the encoded destination url with the email appended', async () => {
+				global[ userDataGlobal ] = userData;
+
+				const testURL = 'https://analytics.google.com/dashboard/';
+				const userEmail = global[ userDataGlobal ].user.email;
+
+				registry.select( CORE_USER ).getAccountChooserURL( testURL );
+
+				await subscribeUntil(
+					registry,
+					() =>
+						registry
+							.select( CORE_USER )
+							.getAccountChooserURL( testURL ) !== undefined
+				);
+
+				const accountChooserURL = registry
+					.select( CORE_USER )
+					.getAccountChooserURL( testURL );
+
+				expect( accountChooserURL ).toBe(
+					`https://accounts.google.com/accountchooser?continue=${ encodeURIComponent(
+						testURL
+					) }&Email=${ encodeURIComponent( userEmail ) }`
+				);
+			} );
+
+			it( 'should return undefined when no data is available', async () => {
+				expect( global[ userDataGlobal ] ).toEqual( undefined );
+				const testURL = 'test destination url';
+
+				const userInfo = registry
+					.select( CORE_USER )
+					.getAccountChooserURL( testURL );
```

Comment:
Technically this test would pass even with a defined user data global, because the first time the user selects this data the selector will return `undefined`. This test should `await` resolution of the `select( CORE_USER ).getUser()` selector, eg:

```js
await subscribeUntil( registry, () =>
	registry
		.select( CORE_USER )
		.hasFinishedResolution( 'getUser' )
);
```